packages/block-editor/src/store/actions.js

```diff
@@ -1282,3 +1283,55 @@ export function setHasControlledInnerBlocks(
 		clientId,
 	};
 }
+
+// This should not be here probably, just experimenting.
+
+/**
+ * Returns an action object used in storing tools panel states for specific blocks.
+ *
+ * @param {string}  blockName Name of the block.
+ * @param {?string} panelName Name of the ToolsPanel/block support
+ */
+
+export const setToolsPanelState = (
+	blockName,
+	panelName,
+	panelItemState
+) => ( { registry } ) => {
+	const existingState =
+		registry
+			.select( preferencesStore )
+			.get( 'core/block-inspector', 'toolsPanel' ) ?? {};
```

Comment:
Most of the other scopes are editor names (`core/edit-post`). What do you think about just using `core` for the first param to represent a cross-editor preference?

Reply:
> Most of the other scopes are editor names (core/edit-post). 

Oh yes, I _was_ wondering about that. Thanks for confirming. 

> What do you think about just using core for the first param to represent a cross-editor preference?

I think "yes" 😄

Reply:
From memory, the block support `ToolsPanel` instances within the site editor have all their controls flagged as defaults (i.e. they'll always be shown). Would that reduce the scope of these preferences to only the block/post editor?

Reply:
You can set different defaults for each editor, but still to keep the scope to just `core`. There's an API for defaults (`dispatch( preferencesStore ).setDefaults`) which is dispatched when each editor is initalized.

Usually the value passed to `setDefaults` mirrors the preference's actual state, but for this case you wouldn't want to list each panel with it's default value, so maybe using a different key for the default could work, e.g.

```js
dispatch( preferencesStore ).setDefaults( { // ... any other preferences, toolsPanelDefault: true } );
```

and then the selector or hook could fall back to that default.

Reply:
My previous comment can be ignored. I had tunnel vision around Global Styles which makes every control displayed by default (no option to toggle display on/off).